Assets/Scripts/Player/PlayerMovementController.cs

```diff
@@ -6,99 +6,117 @@ namespace Code.Scripts
     // Controller class for the player to allow movement of the player model in the demo scene. (Mostly taken from Lab1)
     public class PlayerMovementController : MonoBehaviour
     {
+        [SerializeField] private Transform camera;
+        
         public float acceleration = 1.5f;
         public float decelerationFactor = 1.5f;
         public float maxVelocity = 2.0f;
+        public float angularSmoothTime = 0.1f;
         
         private static readonly int VelocityXHash = Animator.StringToHash("Velocity X");
         private static readonly int VelocityZHash = Animator.StringToHash("Velocity Z");
         
-        private float _velocityX, _velocityZ;
-        private bool _moveForward, _moveBackward, _moveLeft, _moveRight;
+        private float _angularSmoothVelocity;
+        private Vector3 _velocity;
+        private bool _moveForward, _moveBackward, _moveLeft, _moveRight, _freeLook;
         private Animator _animator;
 
         private const KeyCode MoveForwardKey = KeyCode.W;
         private const KeyCode MoveBackwardKey = KeyCode.S;
         private const KeyCode MoveRightKey = KeyCode.D;
         private const KeyCode MoveLeftKey = KeyCode.A;
```

Comment:
It might be a good idea to put the direction keycodes in a struct so that when we have to mess with the controls using the powerup beers, we can just pass the struct as a parameter

Reply:
Done